src/AdvancedHMC.jl

```diff
@@ -34,13 +34,35 @@ include("integrator.jl")
 export Leapfrog, JitteredLeapfrog, TemperedLeapfrog
 
 include("trajectory.jl")
-@deprecate find_good_eps find_good_stepsize
-export EndPointTS, SliceTS, MultinomialTS, 
-       StaticTrajectory, HMCDA, NUTS, 
-       ClassicNoUTurn, GeneralisedNoUTurn, 
-       StrictGeneralisedNoUTurn,
+export Trajectory,
+       FixedNSteps, FixedIntegrationTime,
+       ClassicNoUTurn, GeneralisedNoUTurn, StrictGeneralisedNoUTurn,
+       EndPointTS, SliceTS, MultinomialTS, 
        find_good_stepsize
 
+# Deprecations for trajectory.jl
+
+struct AbstractTrajectory <: AbstractProposal end
+
+struct StaticTrajectory{TS} end
+@deprecate StaticTrajectory{TS}(int::AbstractIntegrator, L) where {TS} Trajectory{TS}(int, FixedNSteps(L))
+@deprecate StaticTrajectory(int::AbstractIntegrator, L) Trajectory{EndPointTS}(int, FixedNSteps(L))
+@deprecate StaticTrajectory(ϵ::AbstractScalarOrVec{<:Real}, L) Trajectory{EndPointTS}(Leapfrog(ϵ), FixedNSteps(L))
+
+struct HMCDA{TS} end
+@deprecate HMCDA{TS}(int::AbstractIntegrator, λ) where {TS} Trajectory{TS}(int, FixedIntegrationTime(λ))
+@deprecate HMCDA(int::AbstractIntegrator, λ) Trajectory{MetropolisTS}(int, FixedIntegrationTime(λ))
+@deprecate HMCDA(ϵ::AbstractScalarOrVec{<:Real}, λ) Trajectory{MetropolisTS}(Leapfrog(ϵ), FixedIntegrationTime(λ))
+
+struct NUTS{TS, TC} end
```

Comment:
@torfjelde Is this what you are asking for?

Reply:
No, I was thinking more like the
```julia
NUTS(int::AbstractIntegrator, args...; kwargs...) = Trajectory{MultinomialTS}(int, GeneralisedNoUTurn(args...; kwargs...))
```
below but _not_ deprecating it. You will inevitably have people coming in that expect a `NUTS` method to exist and will be confused when either
1. It doesn't exists (which should be the case later if we deprecate it, i.e. indicate that in the future versions this will no longer be implemented).
2. There's a deprecation warning.

Reply:
Cool. I will remove the deprecation then.

Reply:
I updated this. I guess this is enough for this PR but we should also add a default for `HMC` in another PR, as requested by Will here: https://github.com/TuringLang/AdvancedHMC.jl/issues/184. Are there other reasonable defaults?